pkg/bloomgateway/worker.go

```diff
@@ -123,9 +124,11 @@ func (w *worker) running(_ context.Context) error {
 			level.Debug(w.logger).Log("msg", "dequeued task", "task", task.ID)
 			w.pending.Delete(task.ID)
 			tasks = append(tasks, task)
+			ctxs = append(ctxs, task.ctx)
 		}
 
-		err = p.run(taskCtx, tasks)
+		multiCtx := multicontext.WithContexts(ctxs...)
```

Comment:
Do we want all requests to cancel if one is? I think that's what's happening here

Reply:
Up for discussion. The other option is to use `taskCtx`, which is not related to any task and cannot be cancelled.

Reply:
My expectation would be that all ongoing child operations for a given task are canceled if the ctx of the given task is canceled. IIUC, with multicontext we are doing the other way around: a task ctx canceled cancels the contexes of other tasks.